.github/actions/tag-deployment/tag-deployment.sh

```
@@ -0,0 +1,167 @@
+    log_warn "No existing tags found with prefix '${TAG_PREFIX}'"
+    latest_tag="${TAG_PREFIX}${INITIAL_VERSION}"
+    echo "$latest_tag"
+    return
+  fi
+
+  echo "$latest_tag"
+}
+
+extract_version() {
+  local tag="$1"
+  echo "${tag#${TAG_PREFIX}}"
```

Shellcheck (0.11.0)

[info] Expansions inside ${..} need to be quoted separately, otherwise they match as patterns.

(SC2295)